system-contracts/test/shared/utils.ts

```diff
@@ -167,3 +172,73 @@ export async function setConstructingCodeHash(address: string, bytecode: string)
   bytecodeHash[1] = 1;
   await accountCodeStorage.storeAccountConstructingCodeHash(address, bytecodeHash);
 }
+
+export interface StateDiff {
+  key: BytesLike;
+  index: number;
+  initValue: BigNumber;
+  finalValue: BigNumber;
+}
+
+export function encodeStateDiffs(stateDiffs: StateDiff[]): string {
+  const rawStateDiffs = [];
+  for (const stateDiff of stateDiffs) {
+    rawStateDiffs.push(
+      ethers.utils.solidityPack(
+        ["address", "bytes32", "bytes32", "uint64", "uint256", "uint256", "bytes"],
+        [
+          ethers.constants.AddressZero,
+          ethers.constants.HashZero,
+          stateDiff.key,
+          stateDiff.index,
+          stateDiff.initValue,
+          stateDiff.finalValue,
+          "0x" + "00".repeat(116),
+        ]
+      )
+    );
+  }
+  return ethers.utils.hexlify(ethers.utils.concat(rawStateDiffs));
+}
+
+export function compressStateDiffs(enumerationIndexSize: number, stateDiffs: StateDiff[]): string {
+  let num_initial = 0;
+  const initial = [];
+  const repeated = [];
+  for (const stateDiff of stateDiffs) {
+    const addition = stateDiff.finalValue.sub(stateDiff.initValue).add(TWO_IN_256).mod(TWO_IN_256);
+    const subtraction = stateDiff.initValue.sub(stateDiff.finalValue).add(TWO_IN_256).mod(TWO_IN_256);
+    let op = 3;
+    let min = stateDiff.finalValue;
+    if (addition.lt(min)) {
+      min = addition;
+      op = 1;
+    }
+    if (subtraction.lt(min)) {
+      min = subtraction;
+      op = 2;
+    }
+    if (min.gte(BigNumber.from(2).pow(248))) {
```

Comment:
comment on why this `if` is needed and what is 248

Reply:
I'm not the author of this implementation.

Funcitons `encodeStateDiffs `and ` compressStateDiffs` have been taken from file `Compressor.spec.ts` in order to avoid code duplication in the file `L1Messenger.spec.ts`. 

 I would suggest removing that code from Compressor.spec.ts, but I aim to keep the tests atomic, addressing specific issues.

Reply:
fixed and removed code from compressor